charts/sonatype-nexus/templates/deployment-statefulset.yaml

```diff
@@ -27,7 +27,7 @@ spec:
   {{- end }}
   {{- if .Values.deploymentStrategy }}
   strategy:
-{{ toYaml .Values.deploymentStrategy | indent 4 }}
```

Comment:
This is the intended behavior. The overrides for deployment strategy would look something like this: 

deploymentStrategy: 
  rollingUpdate:
    maxSurge: 25%
    maxUnavailable: 25%
  type: RollingUpdate